lua/nvim-tree/diagnostics.lua

```diff
@@ -27,7 +27,7 @@ end
 local function from_nvim_lsp()
   local buffer_severity = {}
 
-  for _, diagnostic in ipairs(vim.diagnostic.get()) do
+  for _, diagnostic in ipairs(vim.diagnostic.get(nil, { severity = M.severity })) do
```

Comment:
This is showing only the diagnostics with that exact severity.

`:help diagnostic-severity` indicates that you will need to set a min level.

Reply:
That's correct I'll the default value to `{ min = vim.diagnostic.severity.HINT }`. But I think it's good to keep it like that here so it becomes compatible with other functions that uses severity as an options as per the docs. Example below. What do you think? ^^

```
1. A single [vim.diagnostic.severity](https://neovim.io/doc/user/diagnostic.html#vim.diagnostic.severity) value:
vim.diagnostic.get(0, { severity = vim.diagnostic.severity.WARN })


2. A table with a "min" or "max" key (or both):
vim.diagnostic.get(0, { severity = { min = vim.diagnostic.severity.WARN } })
```

Reply:
We could simply make it explicit in the config and pass it directly to `get()`

```lua
diagnostics = {
  severity = {
    min = vim.diagnostic.severity.HINT,
    max = vim.diagnostic.severity.ERROR,
  },
}
```

This configuration will be validated at setup time.

Reply:
Perfect ^^ I'll do it like that. Thanks!